bin/helpers/reporterHTML.js

```diff
@@ -0,0 +1,198 @@
+
+    specStats = buildSpecStats(specData.meta);
+    specStatsSpan = `<span class='spec-stats ${specStats.cssClass}'> ${specStats.label} </span>`;
+
+    specMeta = `<div class='spec-meta'> ${specNameSpan} ${specPathSpan} ${specStatsSpan} </div>`;
+    sessionBlocks = '';
+    specData.sessions.forEach((specSession) => {
+
+      sessionStatus = specSession.status;
+      sessionClass = sessionStatus === 'passed' ? 'session-passed' : 'session-failed';
+      sessionStatusIcon = sessionStatus === 'passed' ? "&#10004; " : "&#x2717; ";
+
+      sessionLink = `<a href="${specSession.link}" rel="noreferrer noopener" target="_blank"> ${sessionStatusIcon} ${specSession.name} </a>`;
+
+      sessionDetail = `<div class="session-detail ${sessionClass}"> ${sessionLink} </div>`;
+      sessionBlocks = `${sessionBlocks} ${sessionDetail}`;
+    });
+    specSessions = `<div class='spec-sessions'> ${sessionBlocks} </div>`;
+    specRow = `${specRow} <div class='spec-row'> ${specMeta} ${specSessions} </div>`;
+  });
+
+
+  return `<div class='build-table'> ${specRow} </div>`;
+}
+
+function buildSpecStats(specMeta) {
+  let failedSpecs = specMeta.failed,
+      passedSpecs = specMeta.passed,
+      totalSpecs = specMeta.total,
+      specStats = {};
+
+  if (failedSpecs) {
+    specStats.label = `${failedSpecs}/${totalSpecs} FAILED`;
+    specStats.cssClass = 'spec-stats-failed';
+  } else {
+    specStats.label = `${passedSpecs}/${totalSpecs} PASSED`;
+    specStats.cssClass = 'spec-stats-passed';
+  }
+
+  return specStats;
+}
+
+let reportGenerator = (bsConfig, buildId, args, cb) => {
+  let options = {
+    url: `${config.buildUrl}${buildId}/custom_report`,
+    auth: {
+      user: bsConfig.auth.username,
+      password: bsConfig.auth.access_key,
+    },
+    headers: {
+      'User-Agent': utils.getUserAgent(),
+    },
+  };
+
+  return request.get(options, function (err, resp, body) {
+    let message = null;
+    let messageType = null;
+    let errorCode = null;
+    let build;
+
+    if (err) {
+      message = Constants.userMessages.BUILD_INFO_FAILED;
+      messageType = Constants.messageTypes.ERROR;
+      errorCode = 'api_failed_build_info';
+
+      logger.info(message);
+    } else {
+      try {
+        build = JSON.parse(body);
+      } catch (error) {
+        build = null;
+      }
+    }
+
+    if (resp.statusCode == 299) {
+      messageType = Constants.messageTypes.INFO;
+      errorCode = 'api_deprecated';
+
+      if (build) {
+        message = build.message;
+        logger.info(message);
+      } else {
+        message = Constants.userMessages.API_DEPRECATED;
+        logger.info(message);
+      }
+    } else if (resp.statusCode != 200) {
+      messageType = Constants.messageTypes.ERROR;
+      errorCode = 'api_failed_build_generate_report';
+
+      if (build) {
+        message = `${
+          Constants.userMessages.BUILD_GENERATE_REPORT_FAILED.replace('<build-id>>', buildId)
```

Comment:
`<build-id>>` should this be `<build-id>`

Reply:
Done 👍